sdk/nodejs/tests/runtime/langhost/cases/064.async_components/index.js

```diff
@@ -0,0 +1,34 @@
+// Test the ability to invoke provider functions via RPC.
+
+let assert = require("assert");
+let pulumi = require("../../../../../");
+
+class CustResource extends pulumi.CustomResource {
+    constructor(name, opts) {
+        super("test:index:CustResource", name, {}, opts)
+    }
+}
+
+class CompResource extends pulumi.ComponentResource {
+    constructor(name, opts) {
+        super("test:index:CompResource", name, {}, opts)
+
+        // Kick off async work that actually makes our children.
+        this.asyncInitialize().then(_ => this.registerOutputs());
+    }
+
+    async asyncInitialize() {
```

Comment:
This API feels a bit complicated. One has to add three pieces (`isAsyncConstructed`, `asyncInitialize`, and call it from the constructor). Can we not work off just having `asyncInitialize` defined, and then automatically pick it up?

Reply:
not exactly sure what you mean.  can you give an example?

Reply:
So I'm sketching this in C# and thought of something like

``` csharp
class MyStack : Stack
{
    public async Task Initialize()
    {
        // Create components
    }
}
```

So, it's just one method, and by the virtual of defining it I make my stack (component) async.

Reply:
> So, it's just one method, and by the virtual of defining it I make my stack (component) async.

Feels very magical.  what if you mispell it?  Is reflection how this is found?

What if we had an 'AsyncStack' instead as the base class?

Reply:
Okay, I forgot the `override` keyword. It's C#, so we just implement a method from the base class. See the updated #3618 and #3619. Not sure if that directly translates to TS.